### ISSUE TITLE
Use wrapper functions when modifying vma->vm_flags (Linux kernel > 6.3)

### ISSUE THREAD

**Issue by bfstree** [NVIDIA/gds-nvidia-fs#32]
nvidia-fs build fails on Linux kernel 6.3.x.
```
gds-nvidia-fs/src/nvfs-mmap.c:667:23: error: assignment of read-only member ‘vm_flags’
  667 |         vma->vm_flags |= VM_IO | VM_MIXEDMAP | VM_DONTEXPAND | VM_DONTDUMP | VM_DONTCOPY;
      |                       ^~
```

The issue is documented here: [https://lore.kernel.org/lkml/ZA7x9y60sfGOanHl@kroah.com/T/](https://lore.kernel.org/lkml/ZA7x9y60sfGOanHl@kroah.com/T/)
and here: [https://lore.kernel.org/lkml/20230126193752.297968-1-surenb@google.com/](https://lore.kernel.org/lkml/20230126193752.297968-1-surenb@google.com/)

The source should use `vm_flags_set()`, etc. to set the vm_flags.

**Comment by bfstree** [comment]
This also happens with the latest nvidia-fs (v2.18.3) included in CUDA 12.3.1

**Comment by KiranModukuri** [comment]
This issue is addressed in 2.19.6 release
